sentry-core/src/main/java/io/sentry/core/SentryEnvelopeHeaderAdapter.java

```diff
@@ -25,44 +28,64 @@ public void write(JsonWriter writer, SentryEnvelopeHeader value) throws IOExcept
       writer.value(value.getEventId().toString());
     }
 
-    final SdkInfo sdkInfo = value.getSdkInfo();
-    if (sdkInfo != null) {
-      boolean hasInitAttrs = false;
+    final SdkVersion sdkVersion = value.getSdkVersion();
+    if (sdkVersion != null) {
+      if (hasValidSdkVersion(sdkVersion)) {
+        writer.name("sdk").beginObject();
 
-      if (sdkInfo.getSdkName() != null) {
-        hasInitAttrs = initAttrs(writer, hasInitAttrs);
-        writer.name("sdk_name").value(sdkInfo.getSdkName());
+        writer.name("name").value(sdkVersion.getName());
+        writer.name("version").value(sdkVersion.getVersion());
 
-        // we only add versions if the name is not null
-        if (sdkInfo.getVersionMajor() != null) {
-          hasInitAttrs = initAttrs(writer, hasInitAttrs);
-          writer.name("version_major").value(sdkInfo.getVersionMajor());
-        }
-        if (sdkInfo.getVersionMinor() != null) {
-          hasInitAttrs = initAttrs(writer, hasInitAttrs);
-          writer.name("version_minor").value(sdkInfo.getVersionMinor());
-        }
-        if (sdkInfo.getVersionPatchlevel() != null) {
-          hasInitAttrs = initAttrs(writer, hasInitAttrs);
-          writer.name("version_patchlevel").value(sdkInfo.getVersionPatchlevel());
+        final List<String> integrations = sdkVersion.getIntegrations();
+        if (integrations != null) {
```

Comment:
`l`: We could also check if `integrations` is not empty.
```suggestion
        if (integrations != null && !integrations.isEmpty()) {
```

Reply:
`integrations` is always null or non-empty

Reply:
For now yes, but if we change `SdkVersion` and add a `removeIntegration`, which I doubt we gonna, we could have empty integrations. Anyway, the worst thing that can happen is to have an empty `integrations` in the JSON.